.github/workflows/ci.yml

```diff
@@ -0,0 +1,119 @@
+          cache: true
+
+      - name: ci/check-style
+        run: make check-style
+
+      - name: ci/check-modules
+        run: make check-modules
+
+  test:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+        with:
+          fetch-depth: 0
+          path: src/github.com/mattermost/mattermost-operator
+
+      - name: Setup Go
+        uses: actions/setup-go@6edd4406fa81c3da01a34fa6f6343087c207a568 # v3.5.0
+        with:
+          go-version: "1.19"
+          cache: true
+          cache-dependency-path: src/github.com/mattermost/mattermost-operator/go.sum
+
+      - name: Generate Operator Manifests
+        run: |
+          cd ${GITHUB_WORKSPACE}/src/github.com/mattermost/mattermost-operator
+
+          make clean
+          make operator-sdk
+
+          mkdir -p /tmp/apis/mattermost/v1alpha1
+          cp -R apis/mattermost/v1alpha1/* /tmp/apis/mattermost/v1alpha1
+          mkdir -p /tmp/apis/mattermost/v1beta1
+          cp -R apis/mattermost/v1beta1/* /tmp/apis/mattermost/v1beta1
+          mkdir -p /tmp/config/crd/bases
+          cp -R config/crd/bases/* /tmp/config/crd/bases
+
+          GOPATH=${GITHUB_WORKSPACE} make generate manifests
+          diff /tmp/apis/mattermost/v1alpha1 apis/mattermost/v1alpha1
+          diff /tmp/apis/mattermost/v1beta1 apis/mattermost/v1beta1
+          diff /tmp/config/crd/bases config/crd/bases
+
+      - name: ci/test
+        run: |
+          cd ${GITHUB_WORKSPACE}/src/github.com/mattermost/mattermost-operator
+          make unittest goverall
+
+      - name: ci/test-e2e
+        run: |
+          cd ${GITHUB_WORKSPACE}/src/github.com/mattermost/mattermost-operator
+          ./test/e2e.sh
+        env:
+          K8S_VERSION: v1.22.9
+          IMAGE_NAME: mattermost/mattermost-operator
+          IMAGE_TAG: test
+          KIND_VERSION: v0.17.0
+          SDK_VERSION: v1.0.1
+
+  build:
+    if: ${{ github.event_name == 'pull_request' || github.ref_name  == 'master' }}
+    runs-on: ubuntu-latest
+    needs: [lint, test]
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@dc323e67f16fb5f7663d20ff7941f27f5809e9b6 # v3.2.0
+        with:
+          fetch-depth: 0
+
+      - name: Set short SHA
+        run: echo "SHORT_SHA=${GITHUB_SHA:0:7}" >> $GITHUB_ENV
+
+      - name: ci/build-docker
+        run: make build-image
+
+      - name: ci/scan-docker-security
+        uses: aquasecurity/trivy-action@9ab158e8597f3b310480b9a69402b419bc03dbd5 # v0.8.0
+        with:
+          image-ref: "mattermost/mattermost-operator"
+          format: "table"
+          exit-code: "1"
+          ignore-unfixed: true
+          vuln-type: "os,library"
+          severity: "CRITICAL,HIGH"
+
+      - name: ci/push-docker
+        run: |
+          set -e
+          set -u
+          echo $DOCKERHUB_TOKEN | docker login --username $DOCKERHUB_USERNAME --password-stdin
```

Comment:
I'd suggest using a pinned GhAction for docker login https://github.com/docker/login-action and remove bash

Reply:
This is not part of this PR. We are just moving everything as they are, also I believe it's easier to have this repeatable for testing purposes as we have discussed in the past but it's a discussion for a later stage.